build fails with `-Werror=cast-function-type`

I'm trying to build HEAD on my machine; I have pip-installed the requirements and updated the kastore submodule; but running `make ext3` in `python/` results in
```
CFLAGS="-std=c99 -Wall -Wextra -Werror -Wno-unused-parameter -Wno-missing-field-initializers" \
       python3 setup.py build_ext --inplace
running build_ext
building '_tskit' extension
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -std=c99 -Wall -Wextra -Werror -Wno-unused-parameter -Wno-missing-field-initializers -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -UNDEBUG -Ilib -Ilib/subprojects/kastore -I/usr/include/python3.7m -I/usr/lib/python3/dist-packages/numpy/core/include -c _tskitmodule.c -o build/temp.linux-x86_64-3.7/_tskitmodule.o -std=c99
_tskitmodule.c:1851:16: error: cast between incompatible function types from ‘PyObject * (*)(LightweightTableCollection *)’ {aka ‘struct _object * (*)(struct <anonymous> *)’} to ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} [-Werror=cast-function-type]
     {"asdict", (PyCFunction) LightweightTableCollection_asdict,
                ^
_tskitmodule.c:2267:17: error: cast between incompatible function types from ‘PyObject * (*)(IndividualTable *, PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct <anonymous> *, struct _object *, struct _object *)’} to ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} [-Werror=cast-function-type]
     {"add_row", (PyCFunction) IndividualTable_add_row, METH_VARARGS|METH_KEYWORDS,
                 ^
_tskitmodule.c:2278:15: error: cast between incompatible function types from ‘PyObject * (*)(IndividualTable *)’ {aka ‘struct _object * (*)(struct <anonymous> *)’} to ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} [-Werror=cast-function-type]
     {"clear", (PyCFunction) IndividualTable_clear, METH_NOARGS,
               ^
_tskitmodule.c:2695:17: error: cast between incompatible function types from ‘PyObject * (*)(NodeTable *, PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct <anonymous> *, struct _object *, struct _object *)’} to ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} [-Werror=cast-function-type]
     {"add_row", (PyCFunction) NodeTable_add_row, METH_VARARGS|METH_KEYWORDS,
                 ^
_tskitmodule.c:2705:15: error: cast between incompatible function types from ‘PyObject * (*)(NodeTable *)’ {aka ‘struct _object * (*)(struct <anonymous> *)’} to ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} [-Werror=cast-function-type]
     {"clear", (PyCFunction) NodeTable_clear, METH_NOARGS,
```
... and more like this.

I'm probably doing something silly, but can't figure out what it is.

--- a/python/Makefile
+++ b/python/Makefile
@@ -2,7 +2,7 @@
 all: ext3
 
 ext3: _tskitmodule.c 
-	CFLAGS="-std=c99 -Wall -Wextra -Werror -Wno-unused-parameter -Wno-missing-field-initializers" \
+	CFLAGS="-std=c99 -Wall -Wextra -Werror -Wno-unused-parameter -Wno-missing-field-initializers -Wno-cast-function-type" \
 	       python3 setup.py build_ext --inplace
 
 ext2: _tskitmodule.c 

